great_expectations/rule_based_profiler/domain_builder/domain_builder.py

```diff
@@ -64,10 +68,24 @@ def domain_type(self) -> Union[str, MetricDomainTypes]:
     def batch_request(self) -> Optional[Union[BatchRequest, RuntimeBatchRequest, dict]]:
         return self._batch_request
 
+    @batch_request.setter
+    def batch_request(
+        self, value: Union[BatchRequest, RuntimeBatchRequest, dict]
+    ) -> None:
+        self._batch_request = value
+
     @property
     def data_context(self) -> "DataContext":  # noqa: F821
         return self._data_context
 
+    @property
+    def batch(self) -> Optional[Batch]:
+        return self._batch
+
+    @batch.setter
+    def batch(self, value: Batch) -> None:
+        self._batch = value
```

Comment:
❤️

Reply:
@Shinnnyshinshin Happy to discuss what would make sense to document here.  Thanks!